EasyPost/Request.cs

```diff
@@ -53,9 +53,31 @@ public void AddParameter(string name, string? value, ParameterType type)
         /// <param name="parameters">Dictionary of key-value pairs for creating URL query parameters.</param>
         public void AddQueryString(IDictionary<string, object> parameters)
         {
+            if (_restRequest.Method != Method.Get)
+            {
+                throw new ArgumentException("Query string parameters can only be added to GET requests.");
```

Comment:
Please add a checkbox of all langs to our internal task and check off C# is done due to this so we don't forget.